src/check/constrain/generate/call.rs

```diff
@@ -95,22 +95,17 @@ pub fn gen_call(
             property_call(&mut vec![instance.deref().clone()], property, env, ctx, constr)
         }
         Node::Index { item, range } => {
-            generate(range, env, ctx, constr)?;
-
-            let name = Name::from(&HashSet::from([clss::INT, clss::SLICE]));
-            constr.add("index range", &Expected::new(range.pos, &Type { name }), &Expected::from(range), env);
-
-            let (temp_type, helper_ty) = (constr.temp_name(), constr.temp_name());
-            let exp_col = Expected::from(item);
-            let (exp_col1, exp_col2) = Constraint::iterable("index of collection", &exp_col, &temp_type, &helper_ty);
-            constr.add_constr(&exp_col1, env);
-            constr.add_constr(&exp_col2, env);
-
-            let exp_col_ty = Expected::new(ast.pos, &Type { name: temp_type });
-            constr.add("index of collection", &exp_col_ty, &Expected::from(ast), env);
-
-            generate(item, env, ctx, constr)?;
-            Ok(env.clone())
+            let getitem_ast = AST::new(ast.pos, Node::PropertyCall {
```

Comment:
Isn't there already logic for magic methods?
Maybe we can call that here we `GET_ITEM` as argument.